catalog/rest/catalog-rest-endpoint/src/test/java/org/codice/ddf/endpoints/rest/RestEndpointIT.java

```diff
@@ -0,0 +1,249 @@
+        .header(HttpHeaders.CONTENT_LENGTH, is(String.valueOf(contentBytes.length)))
+        .header(HttpHeaders.ACCEPT_RANGES, "bytes")
+        .contentType(contentType)
+        .body(is(contentText));
+  }
+
+  @Test
+  public void testGetInvalidMetacardId() throws Exception {
+    String metacardId = "123";
+
+    Filter filter = mock(Filter.class);
+    QueryResponse queryResponse = mock(QueryResponse.class);
+
+    given(filterBuilder.attribute(Metacard.ID).is().equalTo().text(metacardId)).willReturn(filter);
+    given(catalogFramework.query(any(QueryRequest.class), any(FederationStrategy.class)))
+        .willReturn(queryResponse);
+    given(queryResponse.getResults()).willReturn(ImmutableList.of());
+
+    when()
+        .get(restEndpointUrlBuilder.add(metacardId).build())
+        .then()
+        .assertThat()
+        .statusCode(404)
+        .body(containsString("Unable to retrieve requested metacard."));
+  }
+
+  @Override
+  protected ApplicationOptions getApplicationOptions() {
+    return new DdfComponentOptions(getPortFinder()) {
+
+      @Override
+      protected BundleOption getBundleOptions() {
+        return super.getBundleOptions()
+            .add("org.bouncycastle", "bcprov-jdk15on")
+            .add("ddf.catalog.transformer", "catalog-transformer-attribute");
+      }
+
+      @Override
+      protected FeatureOption getFeatureOptions() {
+        final String[] springFeatures = {
+          "spring-instrument", "spring-jms", "spring-test", "spring-web"
+        };
+
+        final String[] cxfFeatures = {"cxf", "cxf-commands"};
+        final String[] platformFeatures = {"action-core-impl", "commons"};
+
+        return super.getFeatureOptions()
+            .addFeatures("org.apache.karaf.features", "spring", springFeatures)
+            .addFeatures("org.apache.cxf.karaf", "apache-cxf", cxfFeatures)
+            .addFeatures("ddf.platform", "platform-app", platformFeatures)
+            .addFeatureFrom("ddf.thirdparty", "rest-assured", "feature", "rest-assured")
+            .addFeatureFrom(
+                "ddf.platform.util", "util-uuidgenerator-api", "feature", "uuidgenerator-api")
+            .addFeatureFrom("ddf.mime.core", "mime-core-api", "feature", "mime-core-api")
+            .addFeatureFrom("ddf.catalog.core", "catalog-core-api", "feature", "catalog-core-api");
+      }
+    };
+  }
+
+  private void waitForRestEndpoint() {
+    LOGGER.trace("waitForRestEndpoint");
+    with()
+        .pollInterval(FIVE_HUNDRED_MILLISECONDS)
+        .await("REST Endpoint Available")
+        .atMost(TEN_SECONDS)
+        .until(
+            () -> {
+              try {
+                when().head(restEndpointUrlBuilder.build()).then().assertThat().statusCode(200);
+              } catch (Throwable e) {
+                return false;
+              }
+
+              return true;
+            });
+  }
+
+  // TODO - Change to make sure all the bundles of the feature under test are active and move to
```

Comment:
❓ Will you create a ticket for this or address it in this PR?

Reply:
Created https://codice.atlassian.net/browse/DDF-3547 and updated TODO.